drivers/samsung/displays/md_series.cr

```diff
@@ -0,0 +1,500 @@
+  end
+
+  def set_timer(enable : Bool = true, volume : Int32 = 0)
+    # set the time on the display
+    time_request = [] of Int32
+    t = Time.local
+    time_request << t.day
+    hour = t.hour
+    ampm = if hour > 12
+             hour = hour - 12
+             0 # pm
+           else
+             1 # am
+           end
+    time_request << hour
+    time_request << t.minute
+    time_request << t.month
+    year = t.year.to_s(16).rjust(4, '0')
+    time_request << year[0..1].to_i(16)
+    time_request << year[2..-1].to_i(16)
+    time_request << ampm
+
+    do_send("time", time_request)
+
+    state = enable ? "01" : "00"
+    vol = volume.to_s(16).rjust(2, '0')
+    # on 03:45 am enabled  off 03:30 am enabled  on-everyday  ignore manual  off-everyday  ignore manual  volume 15  input HDMI   holiday apply
+    custom_mdc(
+      "A4",
+      "03-2D-01 #{state}  03-1E-01   #{state}      01          80               01           80          #{vol}        21          01"
+    )
+  end
+
+  enum INPUTS
+    Vga           = 0x14 # pc in manual
+    Dvi           = 0x18
+    Dvi_video     = 0x1F
+    Hdmi          = 0x21
+    Hdmi_pc       = 0x22
+    Hdmi2         = 0x23
+    Hdmi2_pc      = 0x24
+    Hdmi3         = 0x31
+    Hdmi3_pc      = 0x32
+    Hdmi4         = 0x33
+    Hdmi4_pc      = 0x34
+    Display_port  = 0x25
+    Dtv           = 0x40
+    Media         = 0x60
+    Widi          = 0x61
+    Magic_info    = 0x20
+    Whiteboard    = 0x64
+  end
+
+  def switch_to(input : String, **options)
+    self[:input_stable] = false
+    self[:input_target] = input
+    do_send("input", INPUTS.parse(input).value, **options)
+  end
+
+  enum SCALEMODE
+    Fill = 0x09
+    Fit =  0x20
+  end
+
+  # Activite the internal compositor. Can either split 3 or 4 ways.
+  def split(inputs : Array(String) = ["hdmi", "hdmi2", "hdmi3"], layout : Int32 = 0, scale : String = "fit", **options)
+    main_source = inputs.shift
+
+    data = [
+      1,                  # enable
+      0,                  # sound from screen section 1
+      layout,             # layout mode (1..6)
+      SCALEMODE.parse(scale).value,  # scaling for main source
+      inputs.flat_map do |input|
+        [INPUTS.parse(input).value, SCALEMODE.parse(scale).value]
+      end
+    ].flatten
+
+    switch_to(main_source, **options)
+    do_send("screen_split", data, **options)
+  end
+
+  def in_range(val : Int32, max : Int32) : Int32
+    min = 0
+    if val < min
+      val = min
+    elsif val > max
+      val = max
+    end
+    val
+  end
+
+  # Emulate mute
+  def mute_audio(val : Bool = true)
+    if val
+      if !self[:audio_mute]?
+        self[:audio_mute] = true
+        self[:previous_volume] = self[:volume].as_i? || 50
+        volume(0)
+      end
+    else
+      unmute_audio
+    end
+  end
+
+  def unmute_audio
+    if self[:audio_mute]?
+      self[:audio_mute] = false
+      volume(self[:previous_volume].as_i? || 50)
+    end
+  end
```

Comment:
More [interfaces](https://github.com/PlaceOS/driver/blob/master/src/placeos-driver/interface/muteable.cr) available for this too.

Reply:
https://github.com/PlaceOS/drivers/pull/19/commits/020b022dc19423575d1c2b045c61bc52b6eb9306